cmd/soroban-cli/src/deploy.rs

```diff
@@ -112,36 +116,59 @@ pub enum Error {
         contract_id: String,
         error: FromHexError,
     },
+    #[error("cannot parse WASM hash {wasm_hash}: {error}")]
+    CannotParseWasmHash {
+        wasm_hash: String,
+        error: FromHexError,
+    },
     #[error("cannot parse secret key")]
     CannotParseSecretKey,
     #[error(transparent)]
     Rpc(#[from] rpc::Error),
+    #[error("either --wasm or --wasm-hash has to be provided")]
+    ContractSourceNotProvided,
+}
+
+enum ContractSource {
+    Wasm(Vec<u8>),
+    WasmHash([u8; 32]),
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
-        let contract = fs::read(&self.wasm).map_err(|e| Error::CannotReadContractFile {
-            filepath: self.wasm.clone(),
-            error: e,
-        })?;
+        let source = if let Some(wasm) = &self.wasm {
+            ContractSource::Wasm(fs::read(wasm).map_err(|e| Error::CannotReadContractFile {
+                filepath: wasm.clone(),
+                error: e,
+            })?)
+        } else {
```

Comment:
this else statement is causing the test to fail. it should just be 
`else if{}`
If you modify this the test can get past that.
https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_else_if